batchflow/research/research.py

```diff
@@ -172,6 +195,26 @@ def add_callable(self, function, *args, returns=None, name=None, execute=1, dump
 
         return self
 
+    def init_instance(self, namespace, name, **kwargs):
+        """ Add namespace to research. Its attributes can be used as callbles and pipelines.
+
+        Parameters
+        ----------
+        namespace : class
+            class which represents namespace.
+        name : str
+            name of the namespace to use in research.
+        """
+        if name in self.namespaces:
+            raise ValueError('Namespace with name {} already exists'.format(name))
+        if name is None:
```

Comment:
Is that branch really needed? I believe that it is better for things as `init_something` to be as explicit as possible, especially regarding the naming, which one will later need to add actions to research